pyproject.toml

```diff
@@ -1,13 +1,16 @@
-[tool.poetry]
+[project]
 name = "pypanther-starter-kit"
+
+[tool.poetry]
 version = "0.1.0"
 description = ""
 authors = ["Panther Labs <pypi@runpanther.io>"]
 readme = "README.md"
+package-mode=false
```

Comment:
Seems the correct mode according to [this](https://python-poetry.org/docs/basic-usage/#operating-modes).

It also fixes the failure reported by CI

Reply:
Hmm, that's a "poetry 2.0" thing. Are we using it?

Reply:
Hmm. Great point. I think we are because this doesn't specify poetry version. 

Maybe I take different approach and
1. Update CI actions to use version 1.*
2. Update docs to reflect that?

Reply:
It's not even correct to install poetry via pip...

I'd suggest:
- install poetry using [the instructions](https://python-poetry.org/docs/1.8/#installation) (I'd suggest installing via pipx)
- pin to latest 1.8 version; saying this because we have some familiarity with it, otherwise it's not wrong to go with 2.0

Reply:
alright. Will make a separate PR for this. Let's pause this review for now and we can come back to it once fixes for poetry are already merged. 

Thanks!

Reply:
Should we break the PR then? And create one for [this](https://github.com/panther-labs/pypanther-starter-kit/pull/48/files#r1913407292)? Or did I misunderstand? 🤔

Reply:
@papanikge  that's what I said I'll do

Reply:
@papanikge @nVlast https://github.com/panther-labs/pypanther-starter-kit/pull/50

Reply:
I was able to run `pypanther upload` successfully after the new changes were applied